degica/komoju-woocommerce
# Dynamic payment methods on checkout

This PR removes the Wordpress level settings to enable/disable payment methods and instead updates the WooCommerce checkout page to dynamically fetch available methods for a merchant using https://github.com/degica/komoju-php. This allows Woocommerce merchants to manage payment methods through their KOMOJU dashboard and have access to all their available methods without configuration. 

It also removes the description from the checkout page that references specific payment methods, and hard coded translations. The translation strings used for payment methods will now be those returned by KOMOJU.

## How to Test
1. Set up the dev environment according to the instructions on this repo
2. On the stage where you configure KOMOJU plugin settings, confirm that on/off checkboxes for payment methods no longer appear as options
3. Navigate to the checkout and confirm that the payment methods listed match those of the secret key configured (note that the plugin communicates with production KOMOJU - you can ask me for help checking this if you can't access the production environment)
4. Create a payment and confirm that you are correctly redirected to the hosted page (an upcoming PR will use the new sessions hosted page, but for now this should be the 'old' page)

Note that a side effect of fetching the methods dynamically is that we can no longer display the "icons" for payment methods unless we choose to manually add assets for them each time. A possible solution to this is returning an asset url from the komoju payment method API. For now these icons have been removed.
